packages/client/src/app/Routes.jsx

```diff
@@ -1,9 +1,36 @@
 import { TabNavigator } from 'react-navigation';
-
+import React from 'react';
+import PropTypes from 'prop-types';
+import { pickBy } from 'lodash';
 import modules from '../modules';
+import { withUser } from '../modules/user/common/containers/AuthBase';
+
+function navTabsFilter(currentUser, currentUserLoading) {
+  // TODO: simplify this by removing 'requiredLogin' param
+  const userFilter = value =>
+    (value.userInfo && value.userInfo.requiredLogin && value.userInfo.role === currentUser.role) ||
+    (value.userInfo && value.userInfo.requiredLogin && !value.userInfo.role) ||
+    !value.userInfo;
+
+  const guestFilter = value => !value.userInfo || (value.userInfo && !value.userInfo.requiredLogin);
+
+  return pickBy(modules.tabItems, currentUser && !currentUserLoading ? userFilter : guestFilter);
+}
+
+class MainScreenNavigator extends React.Component {
```

Comment:
There certainly should be `TODO` here. Routes should work even if `user` module has been removed from the kit

Reply:
They will. If there is no current user logged in 'guestFilter' simply being picked and display all the nav tabs that have not roles restrictions.

Reply:
@evgenijolshanskij Yes, but it is still not right to have auth-specific code in the core of the kit, it should be removed from here and hooks into connectors/here should be added instead to make the approach generic

Reply:
@vlasenko, well, if I got you right the User module should provide its own implementation of the MainScreenNavigator that can be taken instead of the default one.

Reply:
@evgenijolshanskij Yes, thats correct. I'm not sure how to properly solve this problem yet, need to think